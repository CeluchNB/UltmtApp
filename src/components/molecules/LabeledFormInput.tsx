import React from 'react'
import UserInput from '../atoms/UserInput'
import { useColors } from '../../hooks'
import { StyleSheet, Text, View } from 'react-native'
import { size, weight } from '../../theme/fonts'

interface LabeledFormInputProps {
    label: string
    value: any
    keyboardType?: 'default' | 'number-pad'
    unit?: string
    onChange: (...event: any[]) => void
}

const LabeledFormInput: React.FC<LabeledFormInputProps> = props => {
    const { colors } = useColors()
    const {
        children,
        label,
        value,
        keyboardType = 'default',
        unit,
        onChange,
    } = props

    const styles = StyleSheet.create({
        container: {
            display: 'flex',
            flexDirection: 'row',
            alignItems: 'center',
        },
        labelText: {
            fontSize: size.fontLarge,
            fontWeight: weight.bold,
            color: colors.textPrimary,
            width: '50%',
        },
        input: {
            width: unit ? '30%' : '50%',
        },
        unitText: {
            fontSize: size.fontMedium,
            fontWeight: weight.bold,
            color: colors.gray,
            marginLeft: 10,
        },
    })
    return (
        <View style={styles.container}>
            <Text style={styles.labelText}>{label}</Text>
            {children ? (
                children
            ) : (
                <UserInput
                    style={styles.input}
                    placeholder={label}
                    onChangeText={onChange}
                    value={value.toString()}
                    keyboardType={keyboardType}
                />
            )}
            {unit && <Text style={styles.unitText}>{unit}</Text>}
        </View>
    )
}

export default LabeledFormInput
