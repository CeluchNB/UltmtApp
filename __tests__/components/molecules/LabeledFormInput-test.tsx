import LabeledFormInput from '../../../src/components/molecules/LabeledFormInput'
import React from 'react'
import { Text } from 'react-native'
import { fireEvent, render } from '@testing-library/react-native'

const onChange = jest.fn()
const props = {
    label: 'Test field',
    value: '',
    unit: 'players',
    error: 'test error',
    onChange,
}
beforeEach(() => {
    onChange.mockClear()
})

describe('LabeledFormInput', () => {
    beforeAll(() => {
        jest.useFakeTimers()
    })
    afterAll(() => {
        jest.useRealTimers()
    })

    it('should match basic snapshot', () => {
        const snapshot = render(<LabeledFormInput {...props} />)
        expect(snapshot.toJSON()).toMatchSnapshot()
    })

    it('should match snapshot without unit', () => {
        const snapshot = render(
            <LabeledFormInput {...props} unit={undefined} />,
        )
        expect(snapshot.toJSON()).toMatchSnapshot()
    })

    it('should match snapshot with non-user input', () => {
        const snapshot = render(
            <LabeledFormInput {...props}>
                <Text>Test display</Text>
            </LabeledFormInput>,
        )
        expect(snapshot.toJSON()).toMatchSnapshot()
    })

    it('text change should fire prop method', () => {
        const { getByPlaceholderText } = render(<LabeledFormInput {...props} />)

        const input = getByPlaceholderText('Test field')

        fireEvent.changeText(input, 'test')
        expect(onChange).toHaveBeenCalledWith('test')
    })
})
