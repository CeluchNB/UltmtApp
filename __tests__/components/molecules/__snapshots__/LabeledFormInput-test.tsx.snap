// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match basic snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 5,
      "marginTop": 5,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffd600",
        "fontSize": 41.0958904109589,
        "fontWeight": "bold",
        "width": "50%",
      }
    }
  >
    Test field
  </Text>
  <View
    style={
      {
        "width": "30%",
      }
    }
  >
    <View
      style={
        {
          "placeholderTextColor": "#ffffff",
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            {
              "backgroundColor": "#303030",
              "borderColor": "#31adff",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            undefined,
          ]
        }
        testID="text-input-outline"
      />
      <View>
        <View
          style={
            [
              {
                "paddingBottom": 0,
              },
              {
                "minHeight": 56,
                "paddingTop": 8,
              },
            ]
          }
        >
          <TextInput
            autoCapitalize="none"
            editable={true}
            keyboardType="default"
            maxFontSizeMultiplier={1.5}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Test field"
            placeholderTextColor="rgba(73, 69, 79, 1)"
            selectionColor="#ffd600"
            style={
              [
                {
                  "margin": 0,
                  "zIndex": 1,
                },
                {
                  "height": 48,
                },
                {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "paddingHorizontal": 16,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                false,
                [
                  {},
                ],
                undefined,
              ]
            }
            testID="text-input-outlined"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <Text
      style={
        {
          "color": "#ff0000",
          "fontSize": 20.54794520547945,
        }
      }
    >
      test error
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#ababab",
        "fontSize": 30.821917808219176,
        "fontWeight": "bold",
        "marginLeft": 10,
      }
    }
  >
    players
  </Text>
</View>
`;

exports[`should match snapshot with non-user input 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 5,
      "marginTop": 5,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffd600",
        "fontSize": 41.0958904109589,
        "fontWeight": "bold",
        "width": "50%",
      }
    }
  >
    Test field
  </Text>
  <View
    style={
      {
        "width": "30%",
      }
    }
  >
    <Text>
      Test display
    </Text>
    <Text
      style={
        {
          "color": "#ff0000",
          "fontSize": 20.54794520547945,
        }
      }
    >
      test error
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#ababab",
        "fontSize": 30.821917808219176,
        "fontWeight": "bold",
        "marginLeft": 10,
      }
    }
  >
    players
  </Text>
</View>
`;

exports[`should match snapshot without unit 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 5,
      "marginTop": 5,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffd600",
        "fontSize": 41.0958904109589,
        "fontWeight": "bold",
        "width": "50%",
      }
    }
  >
    Test field
  </Text>
  <View
    style={
      {
        "width": "50%",
      }
    }
  >
    <View
      style={
        {
          "placeholderTextColor": "#ffffff",
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            {
              "backgroundColor": "#303030",
              "borderColor": "#31adff",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            undefined,
          ]
        }
        testID="text-input-outline"
      />
      <View>
        <View
          style={
            [
              {
                "paddingBottom": 0,
              },
              {
                "minHeight": 56,
                "paddingTop": 8,
              },
            ]
          }
        >
          <TextInput
            autoCapitalize="none"
            editable={true}
            keyboardType="default"
            maxFontSizeMultiplier={1.5}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Test field"
            placeholderTextColor="rgba(73, 69, 79, 1)"
            selectionColor="#ffd600"
            style={
              [
                {
                  "margin": 0,
                  "zIndex": 1,
                },
                {
                  "height": 48,
                },
                {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "paddingHorizontal": 16,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                false,
                [
                  {},
                ],
                undefined,
              ]
            }
            testID="text-input-outlined"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <Text
      style={
        {
          "color": "#ff0000",
          "fontSize": 20.54794520547945,
        }
      }
    >
      test error
    </Text>
  </View>
</View>
`;
